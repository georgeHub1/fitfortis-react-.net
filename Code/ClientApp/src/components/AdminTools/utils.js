import React from 'react';
import { ReactComponent as FlagBgBg } from '../../img/languages/bg-BG.svg';
import { ReactComponent as FlagEnUs } from '../../img/languages/en-US.svg';
import { ReactComponent as FlagUkUa } from '../../img/languages/uk-UK.svg';
import { ReactComponent as FlagEnIn } from '../../img/languages/en-INTERNATIONAL.svg';

export const getCountryIcon = name => {
  const str = (name) ? name.toLowerCase() : '';

  switch (str) {
    case 'en-us': {
      return <FlagEnUs />;
    }
    case 'bg-bg': {
      return <FlagBgBg />;
    }
    case 'uk-ua': {
      return <FlagUkUa />;
    }
    default: {
      return <FlagEnIn />;
    }
  }
};

export const getValidFormat = data => {
  if (data.language && data.language.includes('uk')) {
    return {
      ...data,
      title: data.titleUkUa,
      description: data.descriptionUkUa
    };
  } else if (data.language && data.language.includes('bg')) {
    return {
      ...data,
      title: data.titleBgBg,
      description: data.descriptionBgBg
    };
  } else if (data.language && data.language.includes('en-us')) {
    return {
      ...data,
      title: data.titleEnUs,
      description: data.descriptionEnUs
    };
  }
  return {
    ...data,
    title: data.titleEn,
    description: data.descriptionEn
  };
};
