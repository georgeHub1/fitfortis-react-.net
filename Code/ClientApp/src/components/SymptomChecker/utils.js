export const removeSelectedItem = (items, selected) => {
	for (const i in selected) {
		const index = items.findIndex(x => x.id === selected[i].id &&  x.name === selected[i].name);

		if (index > -1) {
			items.splice(index, 1);
		}
	}
	return items;
};
