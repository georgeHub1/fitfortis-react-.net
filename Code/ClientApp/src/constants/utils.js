const normalizeMixedDataValue = value => {
  var padding = '000000000000000';

  // Loop over all numeric values in the string and
  // replace them with a value of a fixed-width for
  // both leading (integer) and trailing (decimal)
  // padded zeroes.
  value = value.replace(
    /(\d+)((\.\d+)+)?/g,
    ($0, integer, decimal, $3) => {
      // If this numeric value has "multiple"
      // decimal portions, then the complexity
      // is too high for this simple approach -
      // just return the padded integer.
      if (decimal !== $3) {
        return (padding.slice(integer.length) + integer + decimal);
      }
      decimal = (decimal || '.0');
      return (padding.slice(integer.length) + integer + decimal + padding.slice(decimal.length));
    }
  );
  return (value);
};

export const sortingData = (data, key) => {
  return data.sort((a, b) => {
    // Normalize the file names with fixed-
    // width numeric data.
    var aMixed = normalizeMixedDataValue(a[key]);
    var bMixed = normalizeMixedDataValue(b[key]);

    return aMixed.localeCompare(bMixed);
    // return (aMixed < bMixed ? -1 : 1);
  });
};

export const arrayBufferToBase64 = buffer => {
  var binary = '';
  var bytes = [].slice.call(new Uint8Array(buffer));

  bytes.forEach(b => binary += String.fromCharCode(b));

  return window.btoa(binary);
};

export const  base64ToBlob = (b64Data, contentType, sliceSize) => {
  var byteCharacters = atob(b64Data);
  var byteArrays = [];

  contentType = contentType || '';
  sliceSize = sliceSize || 512;

  for (let offset = 0; offset < byteCharacters.length; offset += sliceSize) {
    const slice = byteCharacters.slice(offset, offset + sliceSize);

    const byteNumbers = new Array(slice.length);

    for (let i = 0; i < slice.length; i++) {
        byteNumbers[i] = slice.charCodeAt(i);
    }

    const byteArray = new Uint8Array(byteNumbers);

    byteArrays.push(byteArray);
  }
  return new Blob(byteArrays, {type: contentType});
};
