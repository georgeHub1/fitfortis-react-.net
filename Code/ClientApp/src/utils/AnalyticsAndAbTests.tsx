import ReactGA from 'react-ga';
import * as firebase from 'firebase/app';
import 'firebase/analytics';
import auth from '../backendServices/auth';

declare const window: any;
const googleAnalyticID = 'UA-158100904-1';
const googleOptimizeID= 'OPT-MTR8NXP';
const optimizeExperimentId = 'Zo7b_2CqSseKSjw2G2Le4Q';


export const firebaseConfig = {
    apiKey: 'AIzaSyA5ioZ-lMHKL1NbKoPo06WcWj3_fFEjplI',
    authDomain: 'fitfortis-20200928.firebaseapp.com',
    databaseURL: 'https://fitfortis-20200928.firebaseio.com',
    projectId: 'fitfortis-20200928',
    storageBucket: 'fitfortis-20200928.appspot.com',
    messagingSenderId: '841736138954',
    appId: '1:841736138954:web:d4c63168018b81546a4433',
    measurementId: 'G-Y1S2CMVBJH'
};

const firebaseApp = firebase.initializeApp(firebaseConfig);

const language = localStorage.getItem('language');

let userIdValue = '';

const firebaseAnalyticsSetUserProperties = () => {
    firebase.analytics().setUserProperties({ language, isAnonymous: !auth.isAuthenticated()});
};

const firebaseAnalyticsLog = (event, parameters) => {
    try {
        firebaseAnalyticsSetUserProperties();
        if (parameters === undefined) parameters = {};
        parameters.appLanguage  = language;
        parameters.userId = userIdValue;
        parameters.isAnonymous = !auth.isAuthenticated();
        firebase.analytics().logEvent(event, parameters);
    } catch (error) {}
};

const firebaseAnalyticUserId = userId => {
    userIdValue = userId;
    try {
        firebase.analytics().setUserId(userId);
    } catch (err) {}
};

const initializeAnalytics = () => {
    ReactGA.initialize(googleAnalyticID);
}

const  initializeAbTests = () =>  {
    ReactGA.ga('require', googleOptimizeID);
    ReactGA.ga('send', 'pageview');
    if (window.dataLayer) {
      window.dataLayer.push({'event': 'optimize.activate'});
    }
}

const checkIsUserType = () => {
    const intervalId = setInterval(() => {
        if (window.google_optimize !== undefined) {
            const variant = window.google_optimize.get(optimizeExperimentId);
            let userType = variant === '1' ? 'AbTestUser' : 'NormalUser';
            localStorage.setItem('userType', userType);
            clearInterval(intervalId);
        }
      }, 100);
}

const abTest_isCreateAccountInRed = () => {
    const userType = localStorage.getItem('userType');
    if (userType) {
        if (userType === 'AbTestUser') {
            return true;
        }
        return false;
    }
}
const abTest_isArticlesInRandomOrder = () => {
    const userType = localStorage.getItem('userType');
    if (userType) {
        if (userType === 'AbTestUser') {
            return true;
        }
        return false;
    }
}

export default {
    googleAnalyticID,
    googleOptimizeID,
    initializeAbTests,
    initializeAnalytics,
    checkIsUserType,
    abTest_isCreateAccountInRed,
    firebaseAnalyticUserId,
    firebaseAnalyticsLog,
    firebaseAnalyticsSetUserProperties,
    abTest_isArticlesInRandomOrder,
};
