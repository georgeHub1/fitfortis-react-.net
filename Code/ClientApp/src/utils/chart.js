export const isSsr = () => (
  !(typeof window !== 'undefined' && window.document && window.document.createElement
    && window.setTimeout)
);

export function isNumber (val) {
  return typeof val === 'number' && !isNaN(val);
}

export const isNumOrStr = value => {
  return isNumber(value) || typeof value === 'string';
};

export const getPotentialPoint = (dateTimes, currentPoint) => {
  if (!dateTimes.length)
    throw new Error('date time should have value');

  const rightPoints = dateTimes.filter(x => x >= currentPoint);
  const leftPoints = dateTimes.filter(x => x <= currentPoint);

  const rightPoint = rightPoints.length ? rightPoints[0] : leftPoints[leftPoints.length - 1];
  const leftPoint = leftPoints.length ? leftPoints[leftPoints.length - 1] : rightPoints[0];

  if (leftPoint === rightPoint)
    return leftPoint;

  const averagePoint = (leftPoint + rightPoint) / 2;

  return (averagePoint >= currentPoint) ? leftPoint : rightPoint;
};
