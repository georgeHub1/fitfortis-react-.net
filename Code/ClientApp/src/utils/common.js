export const createReducer = handlers =>
  (state = null, action) => {
    if (handlers[action.type]) {
      return handlers[action.type](state, action);
    }
    return state;
  };

export const formatDate = date => {
  var monthNames = [ // TODO: need to be localized
    'Jan', 'Feb', 'Mar',
    'Apr', 'May', 'Jun', 'Jul',
    'Aug', 'Sept', 'Oct',
    'Nov', 'Dec'
  ];

  var day = date.getDate();
  var monthIndex = date.getMonth();
  var year = date.getFullYear();

  return `${ monthNames[monthIndex] } ${ day }, ${ year }`;
};

// TODO need simplify this function
export const getQueryVariable = (variable, url) => {
  if (!url) return '';
  const [, queryString] = url.split('?');
  const vars = queryString.split('&');

  for (let i = 0; i < vars.length; i++) {
    // eslint-disable-next-line no-div-regex
    const pair = vars[i].split(/=(.+)/);

    if (decodeURIComponent(pair[0]) === variable) {
      return decodeURIComponent(pair[1]);
    }
  }
};

export function base64StringtoFile (base64String, filename) {
  const  arr = base64String.split(',');
  const mime = arr[0].match(/:(.*?);/)[1];
  const bstr = atob(arr[1]);

  let n = bstr.length;
  const u8arr = new Uint8Array(n);

  while (n--) {
    u8arr[n] = bstr.charCodeAt(n);
  }
  return new File([u8arr], filename, {type: mime});
}

export const arrayBufferToBase64 = buffer => {
  var binary = '';
  var bytes = [].slice.call(new Uint8Array(buffer));

  bytes.forEach(b => binary += String.fromCharCode(b));

  return window.btoa(binary);
};


export function validateWordCount (stringData, maxSize = 350) {
  if (stringData.length > maxSize) {
    stringData = `${stringData.substr(0, maxSize)}`;
    stringData = `${stringData.substr(0, stringData.lastIndexOf(' '))}...`;
  }
  return stringData;
}
