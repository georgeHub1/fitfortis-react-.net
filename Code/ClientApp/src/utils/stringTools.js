/**
 * @param content: string - Html content which one you need to convert into normal string.
 * @param maxSize: number - Size of string. Default maxSize = 200
 * @return string - return normal string based on given maxSize.
 */
export const getShortDescription = (content, maxSize = 200) => {
  const stringText = [];
  const regex = /(<([^>]+)>)/ig;
  const regextForParagraph = /.*?(?=\<p\>).*?(?=\<\/p\>)/ig;

  // Get App <p> tag from content
  content.replace(regextForParagraph, (match, g1) => {
    const str = match.replace(regex, '');

    stringText.push({ value: str, length: str.length});
  });

  if (stringText.length === 0) {
    content = content.replace(/\n/g, '');
    content.replace(regextForParagraph, (match, g1) => {
      const str = match.replace(regex, '');

      stringText.push({ value: str, length: str.length});
    });
  }


  let lastTrimmedString = (stringText.length > 0) ? stringText[0].value : '';

  if (lastTrimmedString.length > maxSize) {
    lastTrimmedString = `${lastTrimmedString.substr(0, maxSize)}`;
    lastTrimmedString = `${lastTrimmedString.substr(0, lastTrimmedString.lastIndexOf(' '))}...`;
  }
  return lastTrimmedString;
};
