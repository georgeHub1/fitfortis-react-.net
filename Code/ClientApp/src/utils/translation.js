export const flattenMessages = (nestedMessages, prefix = '') => {
  return Object.keys(nestedMessages).reduce((messages, key) => {
    const value = nestedMessages[key];
    const prefixedKey = prefix ? `${prefix}.${key}` : key;

    if (typeof value === 'string') {
      messages[prefixedKey] = value;
    } else {
      messages = {...messages, ...flattenMessages(value, prefixedKey)};
    }

    return messages;
  }, {});
};
