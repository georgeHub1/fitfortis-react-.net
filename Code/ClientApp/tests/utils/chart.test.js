import { getPotentialPoint } from '../../src/utils/chart';

const datetimes = [5, 10, 15, 20];
describe('getPotantialPoint for selecting timeLine chart', () => {

  it('should not select new right point(diff of current point and right point is not more than half of left point and nearest right point )', () => {
    const currentPoint = 6;
    const secondPoint = getPotentialPoint(datetimes, currentPoint);

    expect(secondPoint).toEqual(5);
  });
  it('should select new right point(diff of current point and right point is more than half of left point and nearest right point )', () => {
    const currentPoint = 8;
    const secondPoint = getPotentialPoint(datetimes, currentPoint);

    expect(secondPoint).toEqual(10);
  });
  it('should not select new left point(diff of current point and left point is not more than half of left point and nearest right point )', () => {
    const currentPoint = 9;
    const secondPoint = getPotentialPoint(datetimes, currentPoint);

    expect(secondPoint).toEqual(10);
  });
  it('should select new left point(diff of current point and left point is more than half of left point and nearest right point )', () => {
    const currentPoint = 7;
    const secondPoint = getPotentialPoint(datetimes, currentPoint);

    expect(secondPoint).toEqual(5);
  });
  it('if right point does not exist', () => {
    const currentPoint = 21;
    const secondPoint = getPotentialPoint(datetimes, currentPoint);
    expect(secondPoint).toEqual(20);
  });
  it('if left point does not exist', () => {
    const currentPoint = 4;
    const secondPoint = getPotentialPoint(datetimes, currentPoint);
    expect(secondPoint).toEqual(5);
  });
  it('should throw exception if dateTimes is Empty', () => {
    function getValueFromEmptyArray() {
      const currentPoint = 4;
      return getPotentialPoint([], currentPoint);
    }

    expect(getValueFromEmptyArray).toThrowErrorMatchingSnapshot();
  });
});
