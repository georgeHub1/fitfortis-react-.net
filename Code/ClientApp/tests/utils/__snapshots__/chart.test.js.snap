// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPotantialPoint for selecting timeLine chart should throw exception if dateTimes is Empty 1`] = `"date time should have value"`;
